### PR TITLE
fix scm history going out of sync

### PR DESCRIPTION
**Pull request by erezmus**
#### What it does
- Fixes #2922 issue where new git commits don't appear in the history view
- Also fixes duplicate commits appearing in the history view

#### How to test
- open a git repository
- change a file and commit it
- switch to the history view
- the commit should appear at the top

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/scm-extra/src/browser/history/scm-history-widget.tsx [hunk 1]
```diff
@@ -220,20 +220,30 @@ export class ScmHistoryWidget extends ScmNavigableListWidget<ScmHistoryListNode>
...
 
-                    if (options && ((options.maxCount && sourceHistory.length < options.maxCount) || (!options.maxCount && currentCommits))) {
+                    if (options && ((options.maxCount && history.length < options.maxCount) || (!options.maxCount && currentCommits))) {
                         this.hasMoreCommits = false;
                     }
+
+                    // remove the extra commit
                     if (currentCommits.length > 0) {
-                        sourceHistory = sourceHistory.slice(1);
+                        history = history.slice(0, -1);
```
Comment (federicobozzini) [inline packages/scm-extra/src/browser/history/scm-history-widget.tsx]: Why not?

```suggestion
                        history.pop();
```

So `history` could also be a const again.

Comment (erezmus) [reply packages/scm-extra/src/browser/history/scm-history-widget.tsx]: ah good catch

**Review by federicobozzini — APPROVED** [review]
LGTM!

**Review by federicobozzini — APPROVED** [review]
LGTM!

**Pull request merged**